src/extensions/step_handlers/script_handler/src/script_handler.cpp

```diff
@@ -501,30 +504,43 @@ ADUC_Result ScriptHandlerImpl::PrepareScriptArguments(
  */
 ADUC_Result ScriptHandlerImpl::Install(const tagADUC_WorkflowData* workflowData)
 {
-    ADUC_Result result = PerformAction("--action-install", workflowData);
+    ADUC_Result result = PerformAction("install", workflowData);
     return result;
 }
 
-static ADUC_Result ScriptHandler_PerformAction(const std::string& action, const tagADUC_WorkflowData* workflowData)
+/**
+ * @brief Perform a workflow action. If @p prepareArgsOnly is true, only prepare data, but not actually
+ *        perform any action.
+ *
+ * @param action Indicate an action to perform. This can be 'download', 'install',
+ *               'apply', "cancel", and "is-installed".
+ * @param workflowData An object containing workflow data.
+ * @param prepareArgsOnly  Boolean indicates whether to prepare action data only.
+ * @return ADUC_PerformAction_Results The results.
+ */
+ADUC_PerformAction_Results
+ScriptHandler_PerformAction(const std::string& action, const tagADUC_WorkflowData* workflowData, bool prepareArgsOnly)
 {
     Log_Info("Action (%s) begin", action.c_str());
-    ADUC_Result result = { ADUC_GeneralResult_Failure };
+    ADUC_PerformAction_Results results;
+    results.result.ResultCode = ADUC_GeneralResult_Failure;
 
-    std::string scriptFilePath;
-    std::vector<std::string> args;
-    std::string scriptOutput;
     int exitCode = 0;
+    results.commandLineArgs.clear();
 
     if (workflowData == nullptr || workflowData->WorkflowHandle == nullptr)
     {
         Log_Error("Workflow data or handler is null. This is unexpected!");
-        result.ExtendedResultCode = ADUC_ERC_SCRIPT_HANDLER_INSTALL_ERROR_NULL_WORKFLOW;
-        return result;
+        results.result.ExtendedResultCode = ADUC_ERC_SCRIPT_HANDLER_INSTALL_ERROR_NULL_WORKFLOW;
+        return results;
     }
 
+    const char* apiVer = workflow_peek_update_manifest_handler_properties_string(
+        workflowData->WorkflowHandle, HANDLER_PROPERTIES_API_VERSION);
```

Comment:
I don't think we should allow specifying contract version in the handler properties of update metadata, but it should be tied to the version of the apt handler .so that is installed instead.  Reasoning is ALL past versions of every content handler would need to be installed/included with the .deb pkg. Instead, I think only installing the latest version makes sense.